src/main/kotlin/lotto/domain/LottoGame.kt

```diff
@@ -0,0 +1,89 @@
+package lotto.domain
+
+import java.lang.IllegalArgumentException
+import java.lang.NumberFormatException
+import java.lang.StringBuilder
+import java.math.BigDecimal
+import java.util.regex.Pattern
+
+const val COUNT_OF_NUMBERS = 6
+private const val MAX_NUMBER = 45
+private const val MIN_NUMBER = 1
+const val PRICE_OF_LOTTO = 1000
+private const val NUMBER_RANGE = "(4[0-5]|[1-3][0-9]|[1-9])"
+
+class LottoGame(gameMoney: String) {
+    private val lottoNumbers: MutableList<List<Number>> = mutableListOf()
+    private var prizes: MutableList<Lotto> = mutableListOf()
+    private var totalPrizeMoney: Int = 0
+    var count: Int = 0
+        private set
+    var profitRate: Double = 0.0
+        private set
+
+    private var prizeNumbers: List<Number> = listOf()
+
+    init {
+        checkGameMoneyValidation(gameMoney)
+        Prize.values().filter { it.prizeMoney > 0 }
+            .forEach { prizes.add(Lotto(it)) }
+    }
+
+    fun execute(prizeNumberString: String): List<Lotto> {
+        prizeNumbers = checkPrizeNumbersValidation(prizeNumberString)
+        lottoNumbers.forEach { checkMatch(it) }
+        calculateProfitRate()
+        return prizes
+    }
+
+    fun createLotto(): MutableList<List<Number>> {
+        repeat(count) {
+            lottoNumbers.add((MIN_NUMBER..MAX_NUMBER).shuffled().subList(0, COUNT_OF_NUMBERS).sorted())
+        }
+        return lottoNumbers
+    }
+
+    private fun checkMatch(lottoNumbers: List<Number>) {
+        var count = 0
+        lottoNumbers.forEach { if (prizeNumbers.contains(it)) count++ }
+        val prizeMoney = Prize.getPrizeMoney(count)
+        if (prizeMoney > 0) {
+            val selectedLotto = prizes.find { it.getCountOfMatch() == count }
+            selectedLotto!!.addCount()
+            totalPrizeMoney += selectedLotto.getPrizeMoney()
+        }
+    }
+
+    private fun calculateProfitRate() {
+        profitRate = totalPrizeMoney.toBigDecimal()
+            .divide((count * PRICE_OF_LOTTO).toBigDecimal(), 2, BigDecimal.ROUND_HALF_EVEN)
+            .stripTrailingZeros().toDouble()
+    }
+
+    @Throws(NumberFormatException::class)
+    private fun checkGameMoneyValidation(gameMoney: String) {
+        count = gameMoney.toInt() / PRICE_OF_LOTTO
+        require(gameMoney.toInt() >= PRICE_OF_LOTTO) {
+            throw IllegalArgumentException("$PRICE_OF_LOTTO 보다 큰 숫자를 입력해주세요.")
```

Comment:
아래 코멘트에 대해 고민해 봐요.
https://github.com/next-step/kotlin-lotto/pull/1#discussion_r463585709

```suggestion
            "$PRICE_OF_LOTTO 보다 큰 숫자를 입력해주세요."
```